core/src/engine/mutex_test.cpp

```diff
@@ -91,21 +92,27 @@ TYPED_UTEST_P_MT(Mutex, LockPassing, kThreads) {
   const auto test_deadline = engine::Deadline::FromDuration(kTestDuration);
   TypeParam mutex;
 
+  const size_t thread_count =
+      std::is_same_v<TypeParam, engine::SingleWaitingTaskMutex> ? 2 : kThreads;
+
   const auto work = [&mutex] {
     std::unique_lock lock(mutex, std::defer_lock);
     ASSERT_TRUE(lock.try_lock_for(utest::kMaxTestWaitTime));
   };
 
   while (!test_deadline.IsReached()) {
     std::vector<engine::TaskWithResult<void>> tasks;
-    for (size_t i = 0; i < kThreads; ++i) {
+    for (size_t i = 0; i < thread_count; ++i) {
       tasks.push_back(engine::AsyncNoSpan(work));
     }
     for (auto& task : tasks) task.Get();
   }
 }
 
 TYPED_UTEST_P_MT(Mutex, NotifyAndDeadlineRace, 2) {
+  if constexpr (std::is_same_v<TypeParam, engine::SingleWaitingTaskMutex>) {
+    return;
```

Comment:
This looks like a hack, but other ways are more verbose. Okay, I guess.

Reply:
t is a hack. But I agree that the other ways are dramatically worse.